Source/NSStoryboard.m

```diff
@@ -0,0 +1,280 @@
+  BOOL  success = [loader loadModelData: [_transform dataForIdentifier: identifier]
+                      externalNameTable: table
+                               withZone: [self zone]];
+    
+  if (success)
+    {
+      NSMutableArray *seguesToPerform = [NSMutableArray array];
+      NSMutableArray *placeholders = [NSMutableArray array];
+      NSMapTable *segueMap = [_transform segueMapForIdentifier: identifier];
+      NSWindowController *wc = nil;
+      NSViewController *vc = nil;
+      NSWindow *w = nil;
+            
+      FOR_IN(id, o, topLevelObjects)
+        {
+          if ([o isKindOfClass: [NSWindowController class]])
+            {
+              wc = (NSWindowController *)o;
+              [wc _setSegueMap: segueMap];
+              [wc _setTopLevelObjects: topLevelObjects];
+              [wc _setStoryboard: self];
+              [wc _setOwner: NSApp];
+              result = o;
+            }
+          else if ([o isKindOfClass: [NSViewController class]])
+            {
+              vc = (NSViewController *)o;
+              [vc _setSegueMap: segueMap];
+              [vc _setTopLevelObjects: topLevelObjects];
+              [vc _setStoryboard: self];
+              result = o;
+            }
+          else if ([o isKindOfClass: [NSWindow class]])
+            {
+              w = (NSWindow *)o;
+            }
+          else if ([o isKindOfClass: [NSControllerPlaceholder class]])
+            {
+              [placeholders addObject: o];
+            }             
+        }
+      END_FOR_IN(topLevelObjects);
+      
+      // Process action proxies after so we know we have the windowController...
+      FOR_IN(id, o, topLevelObjects)
+        {
+          if ([o isKindOfClass: [NSStoryboardSeguePerformAction class]])
+            {
+              NSStoryboardSeguePerformAction *ssa = (NSStoryboardSeguePerformAction *)o;
+              NSMapTable *mapTable = [_transform segueMapForIdentifier: identifier];
+              NSStoryboardSegue *ss = [mapTable objectForKey: [ssa identifier]];
+              
+              [ssa setSender: result]; // resolve controller here...
+              [ssa setStoryboardSegue: ss];
+              [ssa setStoryboard: self];
+              if ([[ssa kind] isEqualToString: @"relationship"]) // if it is a relationship, perform immediately
+                {
+                  [seguesToPerform addObject: ssa];
+                }
+            }
+        }
+      END_FOR_IN(topLevelObjects);
+
+      // Depending on which kind of controller we have, do the correct thing....
+      if (w != nil && wc != nil)
+        {
+          [wc setWindow: w];
+        }
+
+      // process placeholders...
+      FOR_IN(id, o, placeholders)
+        {
+          NSControllerPlaceholder *ph = (NSControllerPlaceholder *)o;
+          result = [ph instantiate];
+        }
+      END_FOR_IN(placeholders);
+
+      // perform segues after all is initialized.
+      FOR_IN(id, o, seguesToPerform)
```

Comment:
You could save you the type cast here (and in some other places) by already using the correct type in the loop:

 FOR_IN(NSStoryboardSeguePerformAction *, o, seguesToPerform)

Reply:
Done